### DISCUSSION TITLE
Sentry v7 breaking entire js system in Next.js

### DISCUSSION THREAD

**Discussion by Noitidart** [vercel/next.js#38935] [Help] [answered]
Is it possible to make Next.js bundler transpile Sentry package into ES5? 

I upgraded my Sentry to v7 but they say they default `bundle.min.js` to ES6 instead of ES5. I think this is the root cause of breaking my entire JS on older browsers. Others have the same issue here but no solution - https://github.com/getsentry/sentry-javascript/issues/5452 

Sentry in the release notes here say this:

> Default bundle.min.js to ES6 instead of ES5. [ES5 bundles are still available at bundle.es5.min.js](https://github.com/getsentry/sentry-javascript/blob/7.0.0/MIGRATION.md#renaming-of-cdn-bundles). (https://github.com/getsentry/sentry-javascript/pull/4958)

<img width="997" alt="image" src="https://user-images.githubusercontent.com/6372489/180502654-9c982ae5-7efc-4fff-9ae7-d2dbf8414926.png">

 I am using Typescript next template.

**Comment by Babbili** [comment]
@Noitidart are you using Next Js with TypeScript or JavaScript ?
if with TypeScript adjust that in `tsconfig.json` file
```json
{
  "compilerOptions": {
    "target": "es5",   // or "es6"
    
  }
}
```

**Reply by Noitidart** [reply]
Thanks, I'm using it with Typescript. My target is already set to es5 however this still happens. Here is my tsconfig:

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "baseUrl": "../"
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}

```

**Reply by Babbili** [reply]
@Noitidart  then change it to `es6` as Sentry v7 compiling to ES6
```json
{
  "compilerOptions": {
    "target": "es6",
    
  }
}
```

**Reply by Noitidart** [reply]
Thanks very much for your reply.

I need es5 output right? As browsers don't have all the es6 capabilities.

**Reply by Babbili** [reply]
@Noitidart then if you are loading Sentry from CDN this is the one for ES5 in their [Docs](https://docs.sentry.io/platforms/javascript/install/cdn/)
```html
<script
  src="https://browser.sentry-cdn.com/7.7.0/bundle.es5.min.js"
  integrity="sha384-cS2OTFeQyQ67AqPZ5Eu8P/sY+QUBo2bRzUkWMN4cbg1py5fU/xYc+EyU8/gSMQGn"
  crossorigin="anonymous"
></script>
```

**Reply by Noitidart** [reply]
Oh thank you! Is there a way to do this without using CDN? I was hoping to not introduce a script tag. Sincerely appreciate your help!

**Reply by Babbili** [reply]
you're welcome, i havn't seen any clue about npm packages in their Docs or in `migration.md`, it could be a `--flag` while installing v7 via npm, 
but the CDN script would work in `next/head`

**Reply by Noitidart** [reply]
Ah got it, thanks very much. I'll be on the look out to see and share what I learn. As when I import it I get all the typesafety and declaritive imports for maintainability. I'll avoid the CDN solution for now.

**Comment by lobsterkatie** [comment, marked as answer]
See https://github.com/getsentry/sentry-javascript/issues/5452 for further discussion and https://github.com/getsentry/sentry-javascript/pull/5472 for a fix.

**Reply by Noitidart** [reply]
Thank you Sentry team for helping Next.js users stay true to the pitch that got us here of zero configuration!
